integration/tests/basic/evm/opcodes/test_block_timestamp_block_number.py

```diff
@@ -0,0 +1,194 @@
+import random
+import string
+
+import pytest
+
+import allure
+from utils.accounts import EthAccounts
+from utils.models.result import EthGetBlockByHashResult
+from utils.web3client import NeonChainWeb3Client
+
+
+@allure.feature("JSON-RPC validation")
+@allure.story("Verify events")
+@pytest.mark.usefixtures("accounts", "web3_client")
+@pytest.mark.neon_only
+class TestTimestamp:
+    web3_client: NeonChainWeb3Client
+    accounts: EthAccounts
+
+    def test_block_timestamp_call(self, block_timestamp_contracts, json_rpc_client):
+        contract = block_timestamp_contracts[0]
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.getBlockTimestamp().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        assert hex(contract.functions.getBlockTimestamp().call()) == validate_response.result.timestamp
+
+    def test_block_timestamp_event(self, block_timestamp_contracts, json_rpc_client):
+        contract = block_timestamp_contracts[1]
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.call_event().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
+        validate_response = EthGetBlockByHashResult(**response)
+
+        event_logs = contract.events.Result().process_receipt(resp)
+        assert len(event_logs) == 1
+        assert hex(event_logs[0]["args"]["block_timestamp"]) == validate_response.result.timestamp
+
+    def test_block_timestamp_iterative(self, block_timestamp_contracts, json_rpc_client):
+        contract = block_timestamp_contracts[1]
+        sender_account = self.accounts[0]
+
+        tx = self.web3_client.make_raw_tx(sender_account)
+        instruction_tx = contract.functions.call_events_in_loop().build_transaction(tx)
+        resp = self.web3_client.send_transaction(sender_account, instruction_tx)
+        response = json_rpc_client.send_rpc(method="eth_getBlockByHash", params=[resp["blockHash"].hex(), False])
```

Comment:
maybe it's better to rename variables
it is not good to have response and resp in 1 space

Reply:
Fixed ✅